unyt/array.py

```diff
@@ -257,7 +257,13 @@ def _coerce_iterable_units(input_object, registry=None):
         if any([isinstance(o, unyt_array) for o in input_object]):
             ff = getattr(input_object[0], "units", NULL_UNIT)
             if any([ff != getattr(_, "units", NULL_UNIT) for _ in input_object]):
-                raise IterableUnitCoercionError(input_object)
+                ret = []
+                for datum in input_object:
+                    try:
+                        ret.append(datum.in_units(ff.units))
+                    except UnitConversionError:
+                        raise IterableUnitCoercionError(str(input_object))
```

Comment:
This is my one important suggestion here, see https://www.python.org/dev/peps/pep-3134/
```suggestion
                    except UnitConversionError as exc:
                        raise IterableUnitCoercionError(str(input_object)) from exc
```

Reply:
I don’t see how this is any better. It would help in the future if you would explicitly explain why a change would help instead of suggesting that someone read a long technical document to explain why.

Reply:
Sure thing. This produce a different traceback.

### with a bare `raise`
```python
try:
    1/0
except Exception:
    raise ValueError
```
one gets
``
Traceback (most recent call last):
  File "/Users/robcleme/dev/yt-project/yt/t..py", line 2, in <module>
    1/0
ZeroDivisionError: division by zero

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/robcleme/dev/yt-project/yt/t..py", line 4, in <module>
    raise ValueError
ValueError
```
This reads as if something unexpected happened in the except block, but it's not the case: we're raising a different error very much on purpose.

### with a `raise from`
```python
try:
    1/0
except Exception as exc:
    raise ValueError from exc
```

```
Traceback (most recent call last):
  File "/Users/robcleme/dev/yt-project/yt/t..py", line 2, in <module>
    1/0
ZeroDivisionError: division by zero

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/robcleme/dev/yt-project/yt/t..py", line 4, in <module>
    raise ValueError from exc
ValueError
```
This is preferable because the error message conveys the designed intention better. This is exactly the situation `raise from` was intended for

Reply:
I don’t think the second version is any more helpful, sorry. It’s fine as-is.